src/ui/src/builder/composables/useDragToSelect.ts

```diff
@@ -0,0 +1,341 @@
+			isHoveringSelectableArea.value = false;
+			return;
+		}
+
+		const target = ev.target as HTMLElement;
+		const wrapper = wrapperRef.value;
+		if (!wrapper || !wrapper.contains(target)) {
+			isHoveringSelectableArea.value = false;
+			return;
+		}
+
+		const targetEl = target.closest<HTMLElement>("[data-writer-id]");
+		if (targetEl) {
+			if (builderMode.value === "blueprints") {
+				if (!shouldAllowSelectionInBlueprints(target)) {
+					isHoveringSelectableArea.value = false;
+					return;
+				}
+			} else {
+				isHoveringSelectableArea.value = false;
+				return;
+			}
+		}
+
+		const unselectableEl = target.closest<HTMLElement>(
+			"[data-writer-unselectable]",
+		);
+		if (unselectableEl || target.classList.contains("selectionRectangle")) {
+			isHoveringSelectableArea.value = false;
+			return;
+		}
+
+		if (builderMode.value === "blueprints") {
+			if (!isClickOnBlueprintsCanvas(target)) {
+				isHoveringSelectableArea.value = false;
+				return;
+			}
+		}
+
+		isHoveringSelectableArea.value = true;
+	}
+
+	function handleMouseup(ev: MouseEvent) {
+		if (!selectionRect.value.isSelecting) return;
+		if (builderMode.value === "preview") return;
+
+		const { left, top, width, height } = selectionRect.value;
+
+		if (width < MIN_SELECTION_SIZE_PX || height < MIN_SELECTION_SIZE_PX) {
+			cleanupDragSelection();
+			return;
+		}
+
+		const wrapper = wrapperRef.value;
+		if (!wrapper) {
+			cleanupDragSelection();
+			return;
+		}
+
+		const wrapperRect = wrapper.getBoundingClientRect();
+		const selectionRectAbsolute = {
+			left: wrapperRect.left + left,
+			top: wrapperRect.top + top,
+			right: wrapperRect.left + left + width,
+			bottom: wrapperRect.top + top + height,
+		};
+
+		const allComponentEls =
+			wrapper.querySelectorAll<HTMLElement>("[data-writer-id]");
+		const selectedComponents: Array<{
+			componentId: string;
+			instancePath: string;
+		}> = [];
+
+		for (const element of Array.from(allComponentEls)) {
+			if (element.closest("[data-writer-unselectable]")) continue;
+
+			const componentRect = element.getBoundingClientRect();
+
+			if (
+				componentRect.left >= selectionRectAbsolute.left &&
+				componentRect.right <= selectionRectAbsolute.right &&
+				componentRect.top >= selectionRectAbsolute.top &&
+				componentRect.bottom <= selectionRectAbsolute.bottom
+			) {
+				const componentId = element.dataset.writerId;
+				const instancePath = element.dataset.writerInstancePath;
+				if (componentId) {
+					selectedComponents.push({ componentId, instancePath });
+				}
+			}
+		}
+
+		if (selectedComponents.length > 0) {
+			builderManager.setSelection(null);
+			selectedComponents.forEach(({ componentId, instancePath }) => {
+				builderManager.appendSelection(
+					componentId,
+					instancePath,
+					"click",
+				);
+			});
+			justCompletedDragSelection.value = true;
+			setTimeout(() => {
+				justCompletedDragSelection.value = false;
+			}, DRAG_SELECTION_CLICK_DELAY_MS);
```

Comment:
I don't get why do we need to wait ? Maybe you want `nextTick`

```suggestion
      nextTick().then(() => justCompletedDragSelection.value = false);
```

It should be a bit better, but it's always a bad idea to wait an arbitrary amount of time.